_posts/2020-03-16-dask-usage-at-blue-yonder.markdown

```diff
@@ -1,131 +1,131 @@
-in favor of critical ones.
-We use this to re-purpose the resources allocated for production clusters but not utilized the whole time
- and use them for development and staging systems.
-
-### Autoscaling
-
-Autoscaling is a mechanism we implemented to dynamically adapt the number of workers in a Dask cluster
-to the load on the cluster. For this, we added the ``desired_workers`` metric to Distributed, which exposes
-the degree of parallelism that a computation has and allows us to infer how much workers a cluster should
-ideally have. Based on this metric, as well as on the overall resources available and on fairness criteria
-(remember, we run a lot of Dask/Distributed clusters), we add or remove workers to our clusters.
-
-Restarting failed workers for resilience and adding/removing workers for autoscaling requires that the Distributed
-cluster can cope with workers being removed and/or new instances being added. While this is a supported use case, 
-it appeared to us that this seemed not a widely used feature. By making heavy use of it, we were able to uncover
-some bugs and to submit the corresponding fixes.
+  ---
+  layout: single
+  title:  "Dask Usage at Blue Yonder"
+  date:   2019-10-25 10:00:00 +0100
+  tags: technology python data-engineering
+  header:
+    overlay_image: assets/images/tech_gear_banner.jpg
+    overlay_filter: 0.2
+    show_overlay_excerpt: false
+  author: Andreas Merkel
+  author_profile: true
+  ---
+  # Dask Usage at Blue Yonder
+
+  Back in 2017, Blue Yonder started to look into 
+  [Dask/Distributed](https://distributed.dask.org) and how we can leverage it to power
+  our machine learning and data pipelines.
+  In 2019, we started to use it heavily in production for providing demand predicions,
+  recommended order quantities, and price proposals to our customers.
+  Time to take a a look at the way we are using Dask!
+
+  ## Use cases
+
+  First, let's have a look at the use cases we have for Dask. The use of Dask at Blue Yonder
+  is strongly coupled to the concept of [Datasets](https://tech.jda.com/introducing-kartothek/),
+  tabular data stored in blob storage. We use these datasets for storing everything that is needed
+  for generating predictions, recommended order quantities, and recommended prices for our customers,
+  as well as the predictions, order quantities, and prices themselves.
+
+  ### Downloading data from a database to a dataset
+
+  One typical use case that we have is downloading large amounts of data from a database into a
+  dataset. We use Dask/Distributed here as a means to parallelize the download and the graph is
+  an extremely simple: a lot of parallel nodes downloading pieces of data into a blob and one
+  reduction node writing the dataset metadata (which is basically a dictionary what data can be found in which blob).
+
+  ### Doing computations on a dataset
+
+  A lot of our algorithms for machine learning, forecasting, and optimization work on datasets.
```

Comment:
seems redundant to tell

Reply:
Yes, that's true. I was looking for some kind of introductory sentence here.